sdk/python/feast/infra/online_stores/contrib/hazelcast_online_store/hazelcast_online_store.py

```diff
@@ -0,0 +1,230 @@
+import logging
+from datetime import datetime
+from typing import Sequence, List, Optional, Tuple, Dict, Callable, Any, Literal
+import pytz
+
+from feast import RepoConfig, FeatureView, Entity
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from hazelcast import HazelcastClient
+
+from feast.repo_config import FeastConfigBaseModel
+
+from pydantic import StrictStr
+from pydantic.types import StrictBool
+
+from feast.usage import log_exceptions_and_usage
+
+
+class HazelcastOnlineStoreConfig(FeastConfigBaseModel):
+    """Online store config for Hazelcast store"""
+
+    type: Literal["hazelcast"] = "hazelcast"
+    """Online store type selector"""
+
+    cluster_name: StrictStr = "dev"
+    """Name of the cluster you want to connect. The default cluster name is `dev`"""
+
+    cluster_members: Optional[List] = ["localhost:5701"]
+    """List of member addresses which is connected to your cluster"""
+
+    cloud: Optional[StrictBool] = False
+    """Enable if you want to connect Hazelcast Viridian service"""
+
+    discovery_token: Optional[StrictStr]
+    """The discovery token of your Viridian cluster"""
+
+    ssl_enabled: Optional[StrictBool] = False
+    """Enable if you want to use TLS to connect your cluster"""
+
+    ca_path: Optional[StrictStr]
+    """"""
+
+    cert_path: Optional[StrictStr]
+    """"""
+
+    key_path: Optional[StrictStr]
+    """"""
+
+    ssl_password: Optional[StrictStr]
+    """"""
+
+
+class HazelcastOnlineStore(OnlineStore):
+    _client: Optional[HazelcastClient] = None
+
+    def _get_client(self, config: HazelcastOnlineStoreConfig):
+        logger = logging.getLogger("hazelcast")
+        logger.setLevel(logging.ERROR)
+        if not self._client:
+            if config.cloud:
+                self._client = HazelcastClient(
```

Comment:
Since we don't have public `Config()` implementation yet, I had to create cloud enabled and ssl enabled clients separately.